File: 800/2126B/main.cpp
#include<bits/stdc++.h>
using namespace std;
 
int main()
{
    int t;
    cin>>t;
    
    while(t--)
    {
        int n,k;
        cin>>n>>k;
        
        vector<int> a(n);
        for(int i=0;i<n;i++)
            cin>>a[i];
        
        int cnt = 0;
        int curr = 0;
        
        for(int i=0;i<n;i++)
        {
            if(a[i] == 1)
            {
                curr = 0;
            }
            else
            {
                curr++;
                if(curr == k)
                {
                    cnt++;
                    i++;
                    curr = 0;
                }
            }
        }
        
        cout<<cnt<<endl;
    }
    
    return 0;
}

File: 800/2126B/main.java
import java.util.*;

public class main 
{
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);

        int t = sc.nextInt();

        while (t-- > 0) 
        {
            int n = sc.nextInt();
            int k = sc.nextInt();

            int[] a = new int[n];
            for (int i = 0; i < n; i++)
                a[i] = sc.nextInt();

            int cnt = 0;
            int curr = 0;

            for (int i = 0; i < n; i++) 
            {
                if (a[i] == 1)
                    curr = 0;
                else 
                {
                    curr++;
                    if (curr == k) 
                    {
                        cnt++;
                        i++;       // skip next element
                        curr = 0;  // reset counter
                    }
                }
            }

            System.out.println(cnt);
        }
    }
}
